src/compton.c

```diff
@@ -628,10 +628,10 @@ win_build_shadow(session_t *ps, win *w, double opacity) {
     return None;
   }
 
-  shadow_pixmap = XCreatePixmap(ps->dpy, ps->root,
-    shadow_image->width, shadow_image->height, 8);
-  shadow_pixmap_argb = XCreatePixmap(ps->dpy, ps->root,
-    shadow_image->width, shadow_image->height, 32);
+  shadow_pixmap = xcb_generate_id(c);
+  shadow_pixmap_argb = xcb_generate_id(c);
+  xcb_create_pixmap(c, 8, shadow_pixmap, ps->root, shadow_image->width, shadow_image->height);
```

Comment:
Maybe it's better to use `xcb_create_pixmap_checked`, and set `shadow_pixmap`, `shadow_pixmap_argb` to None if create pixmap fails.

Reply:
I am just doing the conversion. The code for `XCreatePixmap` looks like this:
```c
Pixmap XCreatePixmap (
    register Display *dpy,
    Drawable d,
    unsigned int width,
    unsigned int height,
    unsigned int depth)
{
    Pixmap pid;
    register xCreatePixmapReq *req;

    LockDisplay(dpy);
    GetReq(CreatePixmap, req);
    req->drawable = d;
    req->width = width;
    req->height = height;
    req->depth = depth;
    pid = req->pid = XAllocID(dpy);
    UnlockDisplay(dpy);
    SyncHandle();
#ifdef USE_DYNAMIC_XCURSOR
    if (depth == 1)
	_XNoticeCreateBitmap (dpy, pid, width, height);
#endif
    return (pid);
}
```
So, this function already cannot fail.


(for further reference `XAllocID` is some pointer magic that turns into a call to the following function:)
```c
/* _XAllocID - resource ID allocation routine. */
XID _XAllocID(Display *dpy)
{
	XID ret = dpy->xcb->next_xid;
	assert (ret != inval_id);
	dpy->xcb->next_xid = inval_id;
	_XSetPrivSyncFunction(dpy);
	return ret;
}
```

Also, why would CreatePixmap fail? The possible errors according to the protocol docs:
- IDChoice (cannot happen here since `xcb_generate_id()` returns valid IDs)
- Drawable (cannot happen here since the drawable `ps->root` really should be valid)
- Value (happens when width or height are zero)
- Alloc... this might be the only valid error, but since the old code did not check for this either... also, why would an alloc error occur? If the X11 server is out of memory, you got worse problems.

Reply:
You are right about the old code that it didn't check for error. I just think it would be nice if we can handle error properly. 

> If the X11 server is out of memory, you got worse problems.

True. So in that case compton should better exit, rather than ignoring the error.

It's fine if you don't want to do it, I could add the error handling myself later.